### ISSUE TITLE
java.lang.NullPointerException on startup

### ISSUE THREAD

**Issue by gennadiiivanov** [intercom/intercom-android#88]
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.XXX.XXX/com.XXX.android.activity.SplashScreenActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2187)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2236)
       at android.app.ActivityThread.access$800(ActivityThread.java:138)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1199)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5095)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at io.intercom.android.sdk.utilities.DeviceUtils.generateMacId(DeviceUtils.java:66)
       at io.intercom.android.sdk.utilities.DeviceUtils.generateDeviceId(DeviceUtils.java:36)
       at io.intercom.android.sdk.user.DeviceData.generateDeviceData(DeviceData.java:26)
       at io.intercom.android.sdk.api.Api.ping(Api.java:132)
       at io.intercom.android.sdk.Lifecycles.onActivityCreated(Lifecycles.java:75)
       at android.app.Application.dispatchActivityCreated(Application.java:189)
       at android.app.Activity.onCreate(Activity.java:903)
       at com.XXX.android.activity.BaseActivity.onCreate(BaseActivity.java:121)
       at com.XXX.android.activity.SplashScreenActivity.onCreate(SplashScreenActivity.java:40)
       at android.app.Activity.performCreate(Activity.java:5231)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2151)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2236)
       at android.app.ActivityThread.access$800(ActivityThread.java:138)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1199)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5095)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)
       at dalvik.system.NativeStart.main(NativeStart.java)

Attached the screenshot with device models on which it happens:
![screen shot 2015-06-05 at 4 18 59 pm](https://cloud.githubusercontent.com/assets/1785007/8014458/f4165816-0b9e-11e5-9eee-c3d887111aa5.png)
![screen shot 2015-06-05 at 4 19 11 pm](https://cloud.githubusercontent.com/assets/1785007/8014456/f414087c-0b9e-11e5-9e14-0d8589817a94.png)
![screen shot 2015-06-05 at 4 19 24 pm](https://cloud.githubusercontent.com/assets/1785007/8014457/f414de8c-0b9e-11e5-9a4c-214aeef9e7d0.png)

Intercom library version: 1.0.3

**Comment by Harkin** [comment]
Hi @gennadiiivanov 

Looking into this one, is this a bug you can reproduce in-house or are you just getting reports in from a released app?

**Comment by gennadiiivanov** [comment]
@Harkin , I am getting these reports from already released app. Cannot reproduce the issue in-house because don't have mentioned devices to test with.

**Comment by Harkin** [comment]
Unfortunately we're in the same situation here. I'm looking into what might be going wrong on those devices

**Comment by gennadiiivanov** [comment]
> Unfortunately we're in the same situation here. I'm looking into what might be going wrong on those devices, why they're returning null when others aren't

I might found the needed device. Please, let me know if you have any debug build so I could test it.

**Comment by gennadiiivanov** [comment]
> I might found the needed device. Please, let me know if you have any debug build so I could test it.

Tried on all devices we have and couldn't reproduce.

**Comment by Harkin** [comment]
:( I'm putting in the necessary null checks here to avoid this happening again on those devices, we'll release a new version of the library with this fix soon (probably tomorrow)

**Comment by Harkin** [comment]
@gennadiiivanov v1.0.4 shouldn't have this problem, let me know if you come across any other issues

**Comment by mudar** [comment]
@harkin @DaleCantwell 
The issue with `DeviceUtils.generateMacId()` was first reported 2 months ago (5 issues reported here), and the answer was that v0.9.6 fixed it.

I've just had a report of the same crash on Motorola XT1254 running 4.4.4. Will update to v1.0.4, but could you please have a second look at this issue?

```
java.lang.RuntimeException: An error occured while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:300)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
Caused by: java.lang.NullPointerException
       at io.intercom.android.sdk.utilities.DeviceUtils.generateMacId(DeviceUtils.java:66)
       at io.intercom.android.sdk.utilities.DeviceUtils.generateDeviceId(DeviceUtils.java:36)
       at io.intercom.android.sdk.user.DeviceData.generateDeviceData(DeviceData.java:26)
       at io.intercom.android.sdk.api.Api.setGCMPushKey(Api.java:146)
       at io.intercom.android.sdk.RealIntercom.setupGCM(RealIntercom.java:266)
       at com.mypackagename.replaced.utils.IntercomHelper.sendRegistrationIdToBackend(IntercomHelper.java:117)
       at com.mypackagename.replaced.utils.IntercomHelper.access$000(IntercomHelper.java:25)
       at com.mypackagename.replaced.utils.IntercomHelper$1.doInBackground(IntercomHelper.java:129)
       at com.mypackagename.replaced.utils.IntercomHelper$1.doInBackground(IntercomHelper.java:121)
       at android.os.AsyncTask$2.call(AsyncTask.java:288)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)```

**Comment by gennadiiivanov** [comment]
@mudar I confirm that the issue is fixed in v1.0.4. We have a thousands of these crashes in 1.0.3 and have no any similar crashes in v1.0.4.

**Comment by mudar** [comment]
Thanks for the fast reply!
I forgot to mention I was running v1.0.0
